modelling/src/main/java/org/axonframework/modelling/command/AggregateAnnotationCommandHandler.java

```diff
@@ -393,12 +394,14 @@ private AlwaysCreateAggregateCommandHandler(MessageHandlingMember<? super T> han
 
         @Override
         public Object handle(CommandMessage<?> command) throws Exception {
+            AtomicReference<Object> resultReference = new AtomicReference<>();
             Aggregate<T> aggregate = repository.newInstance(() -> {
                 T newInstance = factoryMethod.call();
-                handler.handle(command, newInstance);
+                resultReference.set(handler.handle(command, newInstance));
                 return newInstance;
             });
-            return resolveReturnValue(command, aggregate);
+            Object commandHandlingResult = resultReference.get();
+            return commandHandlingResult != null ? commandHandlingResult : resolveReturnValue(command, aggregate);
```

Comment:
I'd rather resolve return value if the handler is of type `Void`. In other situations, it's fine to return `null`:
```
@CommandHandler
public MySuperAwesomeReturnType handle(MyCommand cmd) {
  if (iDontLikeMyCmd(cmd)) {
    return null;
  }
  return nonNullValue;
}
```

Reply:
Couldn't agree more, so added this just now.

Reply:
I'm wondering, why do we need to "resolve" anything? It's a method, so the return value of that method could simply be used as the result of the command, just like any other command handler.

In the constructor, we didn't have a choice. The result value of a constructor is always the created instance. Since it's pretty bad practice to return an aggregate as a command's result, we added the resolver there.

Reply:
So if I follow you correctly, you wouldn't want to resolve the aggregate identifier if the return type is `void`?